x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/SettingsConfig.java

```diff
@@ -172,6 +211,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (alignCheckpoints != null && (alignCheckpoints.equals(DEFAULT_ALIGN_CHECKPOINTS) == false)) {
             builder.field(TransformField.ALIGN_CHECKPOINTS.getPreferredName(), alignCheckpoints > 0 ? true : false);
         }
+        if (usePit != null && (usePit.equals(DEFAULT_USE_PIT) == false)) {
```

Comment:
This is probably a larger conversation but:
> // do not write default values

Seems like a bad idea as a whole. This means we have to consider older transforms automatically (and unexpectedly) changing their behavior between versions. 

But, I suppose that is a larger idea outside of this PR. To me, being explicit makes more sense.

Reply:
^ Just a passing thought, I know this won't get adjusted in this PR as we want consistency with how settings are handled.

Reply:
for pit we are not changing the default. If we decide to change it: yes we need to ensure the behavior works according to the version the transform has been created. Fortunately this is not the case for pit.

But we have changed defaults and e.g. introduced `dates_as_epoch_millis` and `align_checkpoints`. They work by checking the version in addition to the setting, so e.g. for version < 7.11  `dates_as_epoch_millis` is `true`, after 7.11 it is `false`. Note  `dates_as_epoch_millis` does not exist < 7.11

Update has another component: When it rewrites the config it checks the version and e.g. rewrites a 7.11 transform to a current transform, but sets the setting accordingly.

This works actually. We currently have a tiny glitch for `align_checkpoints`, but that fix is in the makings.

Nevertheless this fits to your other comment: settings are returned as Boolean and can be `null`, so the `null` check has to happen on the implementation.